### DISCUSSION TITLE
[BUG] Get-PnPTenantSite : The remote server returned an error: (401) Unauthorized. Using App Registration

### DISCUSSION THREAD

**Discussion by pdrive29** [pnp/powershell#4577] [General] [answered]
Hello, Hope everyone is doing well, I am trying to get the data using the Get-PnPTenant site using app registration,
I have created the certificate and uploaded it to the application 
![image](https://github.com/user-attachments/assets/828e1a40-d33a-47e0-9d1c-de59a57f308f)

I have applied the relevant Graph API App permissions to the App

![image](https://github.com/user-attachments/assets/9430932f-d8a6-4b10-afa0-2394f2793620)

And even gave the application Share-point Admin and Global Admin Role as well.

I am able to successfully connect to the Sharepoint Admin Center. However I do get warning signs to update the PnP Module despite running the Update-Module multiple times as admin 

![image](https://github.com/user-attachments/assets/44bc7426-ee33-4048-8fd2-a931505564ff)

When I run the Get-PnPTenantID I do get my Tenant ID as the output however when it comes to the running the Get-PnPTenant I get the 401 error

I am running this on my windows machine 

I similar bug was reported https://github.com/pnp/powershell/issues/3549 and the fix was to Set-SPOTenant -DisableCustomAppAuthentication $false 

However in this case since I have to use it to run on other tenants I may not be able to use the above command-let and need to leverage app registrations to get it to work

If any one case point me in the right direction I would be greatful

**Comment by gautamdsheth** [comment, marked as answer]
Hey @pdrive29 , you need to provide SharePoint > Sites full control permissions for this . We used SP API under the hood

**Reply by pdrive29** [reply]
Hi @gautamdsheth thank you so much I finally got the command-let to run should have noticed that sooner.  Again thank you very much

**Reply by JaymeGM** [reply]
@gautamdsheth or @pdrive29 , I still don't understand where exactly you are saying to change.
"SharePoint > Sites full control permissions", would that be within the registered Azure app or somewhere else?

**Reply by pdrive29** [reply]
Hi @JaymeGM in your app registration you will not use the Microsoft Graph but select the Sharepoint Under the Microsoft APIs

![image](https://github.com/user-attachments/assets/e3f63b95-9902-4d83-a3ca-be6847f2f6aa)

Then you would need to provide the Sites full control under the application permission

![image](https://github.com/user-attachments/assets/74c59949-6ff8-4f93-ab52-0a16564a7702)

**Reply by JaymeGM** [reply]
@pdrive29 Now I understand where I was going wrong, in my script I was putting Full permission in "Delegated Permission", but the correct thing for this case is "Application Permission", thank you very much for your help.

**Reply by pdrive29** [reply]
@JaymeGM Glad to have helped :)
